### PR TITLE
storage: add dasd prober

### PR DESCRIPTION
**Pull request by raharper**
Probe for dasd devices.  Use dasdview to extract the information
needed about dasd devices and include them in the probe data.

LP: #1862849

### CODE REVIEW

File: probert/dasd.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+       and the s390x device_id used to uniquely identify the device.
+    """
+    log.debug('Probing DASD devies')
+    dasds = {}
+    if not context:
+        context = pyudev.Context()
+
+    for device in context.list_devices(subsystem='block'):
+        # dasd devices have MAJOR 95
+        if device['MAJOR'] != "94":
```
Comment (mwhudson) [inline probert/dasd.py]: comment and code disagree...

File: probert/dasd.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+log = logging.getLogger('probert.dasd')
+
+DASD_FORMAT = r"^format\s+:.+\s+(?P<format>\w+\s\w+)$"
+DASD_BLKSIZE = r"^blocksize\s+:\shex\s\w+\s+dec\s(?P<blksize>\d+)$"
+
+
+def _search(regex, content, groupkey, flags=None):
+    if flags is None:
+        flags = re.MULTILINE
```
Comment (mwhudson) [inline probert/dasd.py]: No call to this passes flags, so may as well make this unconditional?

File: probert/dasd.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+       'ldl formatted': 'ldl',
+       'not formatted': 'not-formatted',
+    }
+    diskfmt = _search(DASD_FORMAT, dasdview_output, 'format')
+    if diskfmt:
+        return mapping.get(diskfmt.lower())
+
+
+def dasdview(devname):
+    ''' Run dasdview on devname and return dictionary of data.
```
Comment (mwhudson) [inline probert/dasd.py]: It just returns the output?

**Comment by mwhudson** [comment]
Looks sort of sane to me but I'm not really the one to judge. A couple of comments on trivialities.

Are the test changes related to dasds in any way? They look sane but unrelated to me.

**Comment by raharper** [comment]
> Looks sort of sane to me but I'm not really the one to judge. A couple of comments on trivialities.
> 
> Are the test changes related to dasds in any way? They look sane but unrelated to me.

I'll drop.  I had started down a path where I was going to read data from sysfs, in which case having a load_file for unittests was helpful.  However, I ended up only needed run 'dasdview' for the information.

File: probert/dasd.py (hunk 1 above)
Comment (raharper) [reply probert/dasd.py]: Will fix.

File: probert/dasd.py (hunk 2 above)
Comment (raharper) [reply probert/dasd.py]: OK

File: probert/dasd.py (hunk 3 above)
Comment (raharper) [reply probert/dasd.py]: Will fix

Comment (raharper) [reply probert/dasd.py]: Yes, will fix

File: probert/dasd.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+
+def probe(context=None):
+    """Examine all dasd devices present and extract configuration attributes
+
+       This data is useful for determining if the dasd device has been
+       formatted, if so what the block size, the partition layout used
+       and the s390x device_id used to uniquely identify the device.
+    """
+    log.debug('Probing DASD devies')
```
Comment (raharper) [inline probert/dasd.py]: I'll likely add an arch check here to avoid spending any time here unless on s390x.

File: probert/dasd.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        context = pyudev.Context()
+
+    for device in context.list_devices(subsystem='block'):
+        # dasd devices have MAJOR 95
+        if device['MAJOR'] != "94":
+            continue
+        # ignore dasd partitions
+        if 'PARTN' in device:
+            continue
+        dasd_info = get_dasd_info(device)
```
Comment (raharper) [inline probert/dasd.py]: This needs try: accept to handle ValueErrors that may pop up.

**Pull request merged**